### ISSUE TITLE
reject messages with a timestamp < the prev timestamp

### ISSUE THREAD

**Issue by pfraze** [ssbc/secure-scuttlebutt#87]
```
pfraze
domanic: we should refine the order. a feed's internal seq should be respected with higher priority than timestamp

domanic
pfraze, if it doesn't have an increasing timestamp, with respect to itself, that should be invalid

pfraze
domanic: i agree, that should cause feed rejection

pfraze
it should be invalidated locally when `add` is called, so unexpected incidents of non-monotonic growth should be blocked from occuring

pfraze
if it is allowed, it would have to be due to some data corruption or from intentionally trying to break the constraint

pfraze
both are good grounds to invalidate the feed

domanic
pfraze, agree
```

**Comment by dominictarr** [comment]
I traveled back in time and implemented it like that in the first place.

**Comment by dominictarr** [comment]
https://github.com/ssbc/secure-scuttlebutt/blob/master/validation.js#L76-L83

**Comment by dominictarr** [comment]
just occured to me that it would be cool if github links to code where transcludes, and actually displayed those lines on _this_ page. oh well.

**Comment by pfraze** [comment]
> I traveled back in time and implemented it like that in the first place.

haha, though actually i dont think that's it. we need

```js
if(prev && prev.timestamp > msg.timestamp) {
  validateSync.reason = 'timestamp shows after prev msgs' timestamp'
  return false
}
```

is that in there and i dont see?

> just occured to me that it would be cool if github links to code where transcludes, and actually displayed those lines on this page. oh well.

agree

**Comment by dominictarr** [comment]
!!???! I linked you to the wrong code block:
I meant this:
https://github.com/ssbc/secure-scuttlebutt/blob/master/validation.js#L68-L74

it checks the sequence, and then checks the timestamp. maybe we should make it use a <= instead of just < though.

**Comment by pfraze** [comment]
aha, there it is! yeah, i think <= would be right

**Comment by dominictarr** [comment]
okay this is implemented in 8.1.1

**Comment by pfraze** [comment]
:+1:
